### PR TITLE
Cortegeinfo

### PR DESCRIPTION
**Pull request by J4ken**

### CODE REVIEW

File: app/database/database.rb [hunk 1]
```diff
@@ -41,7 +43,8 @@ def get(*args)
   def post(*args)
     begin
       self.class.post(*args)
-    rescue
+    rescue Exception => e
```
Comment (ljukas) [inline app/database/database.rb]: Remove puts

File: app/views/cortege_lineups_management/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<h1>updatera kårtege</h1>
```
Comment (ljukas) [inline app/views/cortege_lineups_management/edit.html.erb]: Capitalize :+1:

File: app/database/areas/cortege_lineup_area.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        body: item_params
+      }
+      put("/api/v1/cortege_lineups/#{id}", options.merge(authenticated_options))
+    end
+
+    def delete_cortege_lineup(id)
+      delete("/api/v1/cortege_lineups/#{id}", authenticated_options)
+    end
+
+    def all_corteges_lineup
```
Comment (ljukas) [inline app/database/areas/cortege_lineup_area.rb]: Shouldn't this be the index funcition? therefore the function name should be "cortege_linup"

File: app/assets/stylesheets/cortege_lineups_management.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the cortege_lineups_managements controller here.
```
Comment (patriksletmo) [inline app/assets/stylesheets/cortege_lineups_management.scss]: There is no content in this file, please remove it.

File: app/controllers/cortege_lineups_controller.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+class CortegeLineupsController < NavigationController
+
+  def index
+    @cortege_lineups = database.all_corteges_lineup
+    unless @cortege_lineups.success?
+      flash[:error] = 'Kunde inte hämta lineup'
```
Comment (patriksletmo) [inline app/controllers/cortege_lineups_controller.rb]: Use `flash.now` for messages where there is no redirect.

Comment (patriksletmo) [inline app/controllers/cortege_lineups_controller.rb]: (otherwise they will appear again)

File: app/controllers/cortege_lineups_management_controller.rb [hunk 6]
```diff
@@ -0,0 +1,74 @@
+class CortegeLineupsManagementController < NavigationController
+
+  def index
+    return if require_login!
+    @cortege_lineups = database.all_corteges_lineup
+
+    unless @cortege_lineups.success?
+      flash[:error] = 'Kunde inte hämta lineup'
```
Comment (patriksletmo) [inline app/controllers/cortege_lineups_management_controller.rb]: Use `flash.now` for messages where there is no redirect.

File: app/controllers/cortege_lineups_management_controller.rb [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    end
+  end
+
+  def new
+    return if require_login!
+    @cortege_lineup={}
+
+  end
+
+  def edit
```
Comment (patriksletmo) [inline app/controllers/cortege_lineups_management_controller.rb]: This action doesn't seem to be used. Please remove it and its template.

File: app/controllers/cortege_lineups_management_controller.rb [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+      flash[:error] = 'Kunde inte skapa lineup'
+      redirect_to action: :new
+    end
+  end
+
+  def update
+    return if require_login!
+    response = database.update_cortege_lineup(params[:id], item_params)
+    if response.success?
+      flash[:success] = 'lineup uppdaterad'
```
Comment (patriksletmo) [inline app/controllers/cortege_lineups_management_controller.rb]: Stor bokstav :)

File: app/controllers/cortege_lineups_management_controller.rb [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+      flash[:error] = 'Kunde inte uppdatera lineup'
+      redirect_to action: :index
+    end
+  end
+
+  def delete
+    return if require_login!
+    response = database.delete_cortege_lineup(params[:id])
+    if response.success?
+      flash[:success] = 'lineup borttagen'
```
Comment (patriksletmo) [inline app/controllers/cortege_lineups_management_controller.rb]: Stor bokstav :)

File: app/database/database.rb [hunk 10]
```diff
@@ -41,7 +43,8 @@ def get(*args)
...
+      puts e
```
Comment (patriksletmo) [inline app/database/database.rb]: Don't print the exception

File: app/helpers/cortege_lineups_helper.rb [hunk 11]
```diff
@@ -0,0 +1,2 @@
+module CortegeLineupsHelper
```
Comment (patriksletmo) [inline app/helpers/cortege_lineups_helper.rb]: Empty file, please remove.

File: app/helpers/cortege_lineups_management_helper.rb [hunk 12]
```diff
@@ -0,0 +1,2 @@
+module CortegeLineupsManagementHelper
```
Comment (patriksletmo) [inline app/helpers/cortege_lineups_management_helper.rb]: Empty file, please remove.

File: app/views/cortege_lineups/index.html.erb [hunk 13]
```diff
@@ -0,0 +1,25 @@
+<% content_for :header_title do %>Kårtegetåget<% end %>
+
+
+<div class="cortege-lineup-container">
```
Comment (patriksletmo) [inline app/views/cortege_lineups/index.html.erb]: Should use the class `container` as well for correct margins.

File: app/views/cortege_lineups_management/_form.html.erb [hunk 14]
```diff
@@ -0,0 +1,36 @@
+<form method="post">
+  <%= hidden_field_tag :authenticity_token, form_authenticity_token %>
+  <div class="container lineupContainer">
+    <div class="row">
+      <div class="col s12 m6">
+        <div class="row">
+          <div class="input-field col s12">
+            <textarea id="name" type="text" class="materialize-textarea" name="item[name]" required><%= @cortege_lineup['name'] %></textarea>
```
Comment (patriksletmo) [inline app/views/cortege_lineups_management/_form.html.erb]: Why textareas and not `<input type="text" [...] />`?

File: app/views/cortege_lineups_management/edit.html.erb (hunk 2 above)
Comment (patriksletmo) [inline app/views/cortege_lineups_management/edit.html.erb]: Please remove this file.

File: app/views/cortege_lineups_management/show.json.jbuilder [hunk 15]
```diff
@@ -0,0 +1 @@
+json.partial! "cortege_lineups_managements/cortege_lineups_management", cortege_lineups_management: @cortege_lineups_management
```
Comment (patriksletmo) [inline app/views/cortege_lineups_management/show.json.jbuilder]: What is this? o_o

File: db/schema.rb [hunk 16]
```diff
@@ -29,4 +29,9 @@
     t.string   "picture"
   end
 
+  create_table "cortege_lineups_management", force: :cascade do |t|
```
Comment (patriksletmo) [inline db/schema.rb]: Why is this here?

File: app/views/cortege_lineups_management/index.json.jbuilder [hunk 17]
```diff
@@ -0,0 +1 @@
+json.array! @cortege_lineups_managements, partial: 'cortege_lineups_management/cortege_lineups_management', as: :cortege_lineups_management
```
Comment (patriksletmo) [inline app/views/cortege_lineups_management/index.json.jbuilder]: What is this? o_o

File: app/database/areas/cortege_lineup_area.rb (hunk 3 above)
Comment (patriksletmo) [inline app/database/areas/cortege_lineup_area.rb]: I disagree. The name should however have the 's' somewhere else (all_cortege_lineups).

Comment (ljukas) [inline app/database/areas/cortege_lineup_area.rb]: That is true. But the name is a bit wonky. Since there is ONE lineup but many corteges, So if you return the cortete_lineup you get the linup which includes all the corteges. So that name would make more sense the all_cortege_lineups since there isn't multiple lineups.

File: app/views/cortege_lineups_management/show.json.jbuilder (hunk 15 above)
Comment (ljukas) [inline app/views/cortege_lineups_management/show.json.jbuilder]: NOBODY KNOWS, NOBODY SEES, IT IS A SECRET!

File: app/views/cortege_lineups_management/_form.html.erb (hunk 14 above)
Comment (J4ken) [inline app/views/cortege_lineups_management/_form.html.erb]: Dunno, it worked :)

Comment (ljukas) [inline app/views/cortege_lineups_management/_form.html.erb]: If there text you input is longer then a sentence a textarea is preferred.

Comment (ljukas) [inline app/views/cortege_lineups_management/_form.html.erb]: Though this is a name field, shoulder there as patrik say be input type="text"

File: app/views/cortege_lineups_management/_form.html.erb [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+            <label for="name">Namn</label>
+          </div>
+          <div class="input-field col s12">
+            <textarea id="description" type="text" class="materialize-textarea" name="item[description]" required><%= @cortege_lineup['description'] %></textarea>
+            <label for="description">Beskrivning</label>
+          </div>
+          <div class="input-field col s12">
+            <textarea id="image" type="text" class="materialize-textarea" name="item[image]" required><%= @cortege_lineup['image'] %></textarea>
```
Comment (ljukas) [inline app/views/cortege_lineups_management/_form.html.erb]: Same as above.

File: app/views/cortege_lineups_management/_form.html.erb [hunk 19]
```diff
@@ -0,0 +1,36 @@
...
+            <label for="image">Länk till bild</label>
+          </div>
+          <div class="input-field col s12">
+            <textarea id="cortege_type" type="text" class="materialize-textarea" name="item[cortege_type]" required><%= @cortege_lineup['cortege_type'] %></textarea>
```
Comment (ljukas) [inline app/views/cortege_lineups_management/_form.html.erb]: Same as above.

File: app/views/cortege_lineups_management/_form.html.erb [hunk 20]
```diff
@@ -0,0 +1,36 @@
...
+            <label for="cortege_type">Typ av kårtege</label>
+          </div>
+          <div class="input-field col s12">
+            <textarea id="order" type="number" class="materialize-textarea" name="item[order]" required><%= @cortege_lineup['order']%></textarea>
```
Comment (ljukas) [inline app/views/cortege_lineups_management/_form.html.erb]: Same as above, should maybe even be a slider or similar number input.
